packages/block-editor/src/components/inserter/index.js

```diff
@@ -102,10 +128,62 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
+		const { hasInserterItems, hasOneAllowedItem } = select( 'core/block-editor' );
 
 		return {
 			hasItems: hasInserterItems( rootClientId ),
+			hasOneAllowedItem: hasOneAllowedItem( rootClientId ),
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, { select } ) => {
+		// eslint-disable-next-line no-restricted-syntax
+		function getInsertionIndex() {
```

Comment:
The original `getInsertionIndex`, in `InserterMenu`, should be made into a selector — unless there's a reason not to.